ndb/tests/unit/test_model.py

```diff
@@ -1988,10 +1988,74 @@ def test__from_base_type(self):
 
 
 class TestStructuredProperty:
+    class Book(model.Model):
+        num_pages = model.IntegerProperty()
+        topic = model.StringProperty()
+
+    def test_constructor_defaults(self):
+        prop = model.StructuredProperty(self.Book)
+        assert prop._modelclass is self.Book
+        # Check that exactly one of the constructor defaults was used.
+        assert prop.__dict__ == {"_modelclass": self.Book}
+
     @staticmethod
-    def test_constructor():
-        with pytest.raises(NotImplementedError):
-            model.StructuredProperty()
+    def test_constructor_repeated_not_allowed():
+        class Foo(model.Model):
+            _has_repeated = True
+            bar = model.StringProperty(repeated=True)
+
+        with pytest.raises(TypeError):
+            model.StructuredProperty(Foo, repeated=True)
+
+    def test__get_for_dict_not_set(self):
+        prop = model.StructuredProperty(self.Book, "book")
+        entity = unittest.mock.Mock(
+            _projection=None, _values={}, spec=("_projection", "_values")
+        )
+        nested_values = prop._get_for_dict(entity)
+        assert nested_values is None
+
+        # Cache is untouched.
+        assert model.Property._FIND_METHODS_CACHE == {}
+
+    def test__get_for_dict_single(self):
+        prop = model.StructuredProperty(self.Book, "book")
+        book_value = unittest.mock.Mock(spec=("_to_dict",))
```

Comment:
Rather than a mock, an instance of `Book` should be used here (and in related places)